cmd/root.go

```diff
@@ -438,7 +478,159 @@ func createIssue(issue github.Issue, client jira.Client) error {
 		return err
 	}
 
-	log.Debugf("Created JIRA issue #%s!", jIssue.ID)
+	log.Debugf("Created JIRA issue %s!", jIssue.Key)
+
+	if err = createComments(issue, *jIssue, nil, ghClient, jClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var jCommentRegex = regexp.MustCompile("^Comment \\(ID (\\d+)\\) from GitHub user (\\w+) \\((.+)\\)? at (.+):\\n\\n(.+)$")
+var jCommentIDRegex = regexp.MustCompile("^Comment \\(ID (\\d+)\\)")
+
+func createComments(ghIssue github.Issue, jIssue jira.Issue, existing []jira.Comment, ghClient github.Client, jClient jira.Client) error {
+	if *ghIssue.Comments == 0 {
+		log.Debugf("Issue #%d has no comments, skipping.", *ghIssue.Number)
+		return nil
+	}
+
+	ctx := context.Background()
+	repo := strings.Split(rootCmdCfg.GetString("repo-name"), "/")
+	comments, _, err := ghClient.Issues.ListComments(ctx, repo[0], repo[1], *ghIssue.Number, &github.IssueListCommentsOptions{
+		Sort:      "created",
+		Direction: "asc",
+	})
+	if err != nil {
+		log.Errorf("Error retrieving GitHub comments for issue #%d. Error: %s.", *ghIssue.Number, err)
+		return err
+	}
+
+	for _, ghComment := range comments {
+		found := false
+		for _, jComment := range existing {
+			if !jCommentIDRegex.MatchString(jComment.Body) {
+				continue
+			}
+			// matches[0] is the whole string, matches[1] is the ID
+			matches := jCommentIDRegex.FindStringSubmatch(jComment.Body)
+			id, _ := strconv.Atoi(matches[1])
+			if *ghComment.ID != id {
+				continue
+			}
+			found = true
+
+			updateComment(*ghComment, jComment, jIssue, ghClient, jClient)
```

Comment:
if you have 20 Jira comments and you find the GitHub comment on the first 1, you will loop over the remaining 19 comments. Instead, you should `break` after updateComment.